src/csharp/Pester/Test.cs

```diff
@@ -36,6 +36,9 @@ public Test()
         public string ExpandedPath { get; set; }
 
         public string Result { get; set; }
+        public string ExpectedValue { get; set; }
+        public string ActualValue { get; set; }
+        public string BecauseValue { get; set; }
```

Comment:
Do they need to be publicly accessible? If so, we'd need something like a single `ShouldResults[]` property. A test can have multiple assertions and could return multiple errors with `Should.ErrorAction = 'Continue'`. The values can be accessed through `ErrorRecord[i].TargetObject` - good enough?

Reply:
The targetobject can be sufficient but I wanted to surface them in an intellisense-friendly manner, and this is the same process the other properties are extracted.

Reply:
True, but they're also test-specific (single valued) and mostly non-customizable. I guess a `AssertionError[] FailedAssertions`-property could work. Should include `TargetObject.Line` to identify which assertion failed.

`Should` only supports 5-6 custom assertions IIRC due to max parameter sets, but in the future we could consider a `Add-AssertionError` function for third-party assertion modules like [Assert](https://github.com/nohwnd/assert) to integrate with it. Let's see what @nohwnd thinks 🙂

Reply:
Apologies, I didn't fully understand what you were communicating, but yes, we could surface the multiple assertion errors as their own structured objects, but I don't think this happens today for failuremessages does it? I was just trying to follow the existing pattern.

Reply:
Fixed 83d0352